src/track/taglib/trackmetadata_ape.cpp

```diff
@@ -97,19 +100,22 @@ void importTrackMetadataFromTag(TrackMetadata* pTrackMetadata, const TagLib::APE
     QString albumArtist;
     if (readItem(tag, "Album Artist", &albumArtist) ||
             readItem(tag, "ALBUM ARTIST", &albumArtist) ||
-            readItem(tag, "ALBUMARTIST", &albumArtist)) {
+            readItem(tag, "ALBUMARTIST", &albumArtist) ||
+            resetMissingTagMetadata) {
         pTrackMetadata->refAlbumInfo().setArtist(albumArtist);
```

Comment:
In order for that to work, `readItem` must guarantee not to touch `albumArtist` when it returns `false`, can we be sure of that?

Reply:
That's how all those read functions work. Otherwise the existing code would not work either.